modules/test/tls/python/src/tls_module.py

```diff
@@ -265,18 +266,60 @@ def _security_tls_v1_3_server(self):
         description = 'TLS 1.3 certificate is valid'
       else:
         description = 'TLS 1.3 certificate is invalid'
-      return results[0], description,results[1]
+      return results[0], description, results[1]
 
     else:
       LOGGER.error('Could not resolve device IP address')
       return 'Error', 'Could not resolve device IP address'
 
+  def _security_tls_v1_0_client(self):
+    LOGGER.info('Running security.tls.v1_0_client')
+    self._resolve_device_ip()
+    # If the ipv4 address wasn't resolved yet, try again
+    if self._device_ipv4_addr is not None:
+      tls_1_0_valid = self._validate_tls_client(self._device_ipv4_addr, '1.0')
+      tls_1_1_valid = self._validate_tls_client(self._device_ipv4_addr, '1.1')
+      tls_1_2_valid = self._validate_tls_client(self._device_ipv4_addr, '1.2')
+      tls_1_3_valid = self._validate_tls_client(self._device_ipv4_addr, '1.3')
+      states = [
+          tls_1_0_valid[0], tls_1_1_valid[0], tls_1_2_valid[0], tls_1_3_valid[0]
+      ]
+      if any(state is True for state in states):
+        # If any state is True, return True
+        result_state = True
+      elif all(state == 'Feature not Detected' for state in states):
+        # If all states are "Feature not Detected"
+        result_state = 'Feature not Detected'
+      elif all(state == 'Error' for state in states):
+        # If all states are "Error"
+        result_state = 'Error'
+      else:
+        result_state = False
+      LOGGER.info(f'Result State: {result_state}')
+      if result_state:
+        result_message = 'TLS 1.0 or higher detected'
+      else:
+        result_message = 'TLS 1.0 or higher was not detected'
+      result_details = tls_1_0_valid[2] + tls_1_1_valid[2] + tls_1_2_valid[
+          2] + tls_1_3_valid[2]
+      result_tags = []
+      result_tags.extend(tls_1_0_valid[3])

```

Comment:
Is this adding multiple tags to the result?

Reply:
Good catch, updated to prevent duplicate entries.